src/app.tsx

```diff
@@ -1,171 +1,147 @@
-        <ErrorBoundary>
-          <ThemeProvider theme={YCAITheme}>
-            <YTContributionInfoBox />
-          </ThemeProvider>
-        </ErrorBoundary>
-      </React.StrictMode>,
-      document.getElementById('ycai-contribution-box')
-    );
-  }
+  React.useEffect(() => {
 
-  // video recommendations box
-  if (settings?.enhanceYouTubeExperience === true) {
+    // append yt video page recommendations dom element
     if (ytRelatedVideoNode !== null) {
       appLogger.debug('Element (%s) found in yt page', ytRelatedVideoNode.id);
-      const ycRecommendationsEl = ytRelatedVideoNode.querySelector(
-        YC_RECOMMENDATIONS_SELECTOR
-      );
 
-      if (ycRecommendationsEl !== null) {
+      if (ycRecommendationsNode === null) {
         appLogger.debug(
-          'Element (%s) present. Returning...',
+          'Element (%s) not found in dom, attaching it..',
           YC_RECOMMENDATIONS_SELECTOR
         );
-        return undefined;
+        const ycRelatedNode = document.createElement('div');
+        ycRelatedNode.id = YC_RECOMMENDATIONS_ID;
+        ytRelatedVideoNode.prepend(ycRelatedNode);
       }
-
       appLogger.debug(
-        'Element (%s) not found in dom, attaching it..',
+        'Element (%s) present. Returning...',
         YC_RECOMMENDATIONS_SELECTOR
       );
-      const ycMainNode = document.createElement('div');
-      ycMainNode.id = YC_RECOMMENDATIONS_ID;
-      ytRelatedVideoNode.prepend(ycMainNode);
-
-      ReactDOM.render(
-        <React.StrictMode>
-          <ErrorBoundary>
-            <ThemeProvider theme={YCAITheme}>
-              <YTVideoPage />
-            </ThemeProvider>
-          </ErrorBoundary>
-        </React.StrictMode>,
-        document.getElementById(YC_RECOMMENDATIONS_ID)
-      );
     }
-  } else {
-    appLogger.debug('Removing element (%s)', YC_RECOMMENDATIONS_SELECTOR);
-    document.querySelector(YC_RECOMMENDATIONS_SELECTOR)?.remove();
-    document.querySelector(YC_CONTRIBUTION_INFO_BOX_SELECTOR)?.remove();
-  }
-  return undefined;
-};
-
-/**
- * Define mutation observer to listen for window's dom changes
- */
-void pipe(
-  settingsRefetch.run(),
-  TE.map((settings) => {
-    if (settings.active) {
-      if (settings.independentContributions.enable) {
-        appLogger.debug(
-          'Independent contribution enabled. Getting the keypair...'
-        );
 
-        const observer = new MutationObserver(
-          debounce(
-            (mutations) => {
-              appLogger.debug(`Mutations received %O`, mutations);
-              renderInjectedElements(settings);
-            },
-            2500,
-            // NOTE: 2500ms to collect evidence as
-            // don't need that much frequency.
-            { trailing: true }
-          )
-        );
-
-        observer.observe(window.document.body, {
-          subtree: true,
-          childList: true,
-        });
+    // append yc contribution box
+    if (ycContributionInfoBoxNode === null) {
```

Comment:
can we hide this for now? maybe behind the `independentContributions.showUI` flag because it's useful for debugging but in terms of UX, as is, it is difficult to understand